insights/parsers/mysqladmin.py

```diff
@@ -0,0 +1,132 @@
+"""
+mysqladmin command - Command
+============================
+
+Parsing and extracting data from output of command ``/bin/mysqladmin variables``.
+Parsers contained in this module are:
+
+MysqladminVars - command ``/bin/mysqladmin variables``
+------------------------------------------------------
+
+"""
+
+from .. import Parser, parser
+from insights.parsers import ParseException
+from insights.specs import Specs
+
+
+@parser(Specs.mysqladmin_vars)
+class MysqladminVars(Parser):
+    """
+    The output of command ``/bin/mysqladmin variables`` is in mysql table format,
+    contains 'Variable_name' and 'Value' two columns.
+    This parser will parse the table and set each variable as an class attribute.
+
+    Sample command output::
+
+        +-------------------------------------------------+------------------+
+        | Variable_name                                   | Value            |
+        +-------------------------------------------------+------------------+
+        | aria_block_size                                 | 8192             |
+        | aria_checkpoint_interval                        | 30               |
+        | aria_checkpoint_log_activity                    | 1048576          |
+        | datadir                                         | /var/lib/mysql/  |
+        | version                                         | 5.5.56-MariaDB   |
+        +-------------------------------------------------+------------------+
+
+
+    Example:
+        >>> output.version
+        '5.5.56-MariaDB'
+        >>> 'DataDir' in output
+        True
+        >>> output.get('what', '233')
+        '233'
+        >>> config.getint('aria_block_size', '4096')
+        8192
+    """
+    def parse_content(self, content):
+        """
+        Parse output content table of command ``/bin/mysqladmin variables``.
+        Set each variable as an class attribute.
+        """
+        if len(content) < 5:
+            raise ParseException("Empty or wrong content in table.")
+
+        self._data = {}
+        for l in content[3:-1]:
+            lsp = l.split('|')
+            if len(lsp) != 4:
+                raise ParseException("Unparseable line in table.")
+            self._data[lsp[1].strip().lower()] = lsp[2].strip()
+
+        for k, v in self._data.items():
```

Comment:
This syntactic sugar could be useful, but to use it, you have to check for the key's existence to avoid an AttributeError being raised which reduced the value of the functionality. Better just fix several common attributes (e.g version, datadir etc.) and explicitly document them in the docstring.